Changes and additions

wilecoyote opened the pull request:
**Changed files:**
* main.lua, cheat_core_buffs.lua, cheat_core_player.lua and buff_cheat.xml

**Changes in detail:**
* Changed version in main.lua to 1.53.
* Changed invisibility and immortal buff to be cheat specific.
* Added a lot of teleport commands (bandit camps, bath houses, charcoal-burner camps, woodland gardens, herbalists, mills, taverns, and woodcutter camps)
* Added locations for bath houses, charcoal-burner camps, herbalists, mills, taverns, and woodcutter camps.
* Removed doubled entries for treasure chests.

wilecoyote committed:
Merge branch 'master' of https://github.com/wilecoyote/kcd-cheat

wilecoyote committed:
Made buffs cheat mod specific

wilecoyote committed:
Corrected typo

wilecoyote committed:
Corrected a copy and paste mistake

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -108,147 +87,336 @@ function cheat:teleport_to(line)
+}
 
+cheat:createCommand("cheat_tp_bc", "cheat:tp_bc(%line)", cheat.cheat_tp_bc_args,
+"Teleports the player to the given place. Supported places (case insensitive):\n$8RN_1, RN_2, RN_3, RN_4, RS_1, RS_2, RS_3, RS_4, RS_5, IL_1, IL_2, IL_3,\n$8BC_1, BC_2, BC_3, BC_4, BC_5, BC_6, BC_7, BC_8, BC_9, BC_10",
```
Should change 1st sentence. "Teleports the player to the given bandit camp." so people know what this does.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -108,147 +87,336 @@ function cheat:teleport_to(line)
+}
+
+cheat:createCommand("cheat_tp_bh", "cheat:tp_bh(%line)", cheat.cheat_tp_bh_args,
+"Teleports the player to the given place. Supported places (case insensitive):\n$8LEDETCHKO, RATTAY, SASAU, TALMBERG",
```
Should change 1st sentence. "Teleports the player to the given bath house." so people know what this does.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -108,147 +87,336 @@ function cheat:teleport_to(line)
+  places["LEDETCHKO"] = "x:2042 y:1282 z:28"
+  places["RATTAY"] = "x:2380 y:595 z:31"
+  places["SASAU"] = "x:959 y:1317 z:19"
+  places["TALMBERG"] = "x:2390 y:2810 z:89"  
```
Is it worth just numbering the keys now that each location type has its own command?
ex: cheat:tp_bh place:1

wilecoyote replied:
Good point. But what should the description look like then? LEDETCHKO = 1, RATTAY = 2 or LEDETCHKO (1), RATTAY (2) or (LEDETCHKO) 1, (RATTAY) 2?

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -108,147 +87,336 @@ function cheat:teleport_to(line)
+}
+
+cheat:createCommand("cheat_tp_cb", "cheat:tp_cb(%line)", cheat.cheat_tp_cb_args,
+"Teleports the player to the given place. Supported places (case insensitive):\n$8NH_NORTH, NH_SOUTH, RATTAY, ROVNA, SASAU, TALMBERG",
```
Suggest same fix here as well.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -108,147 +87,336 @@ function cheat:teleport_to(line)
+}
+
+cheat:createCommand("cheat_tp_gn", "cheat:tp_gn(%line)", cheat.cheat_tp_gn_args,
+"Teleports the player to the given place. Supported places (case insensitive):\n$8GN_1, GN_2, GN_3, GN_4, GN_5, GN_6, GN_7, GN_8, GN_9, GN_10, GN_11, GN_12, GN_13, GN_14, GN_15,\n$8GN_16, GN_17, GN_18, GN_19, GN_20, GN_21, GN_22, GN_23, GN_24, GN_25, GN_26, GN_27, GN_28, GN_29, GN30, GN_31, GN_32, GN_33, GN_34, GN_35, GN_36, GN_37, GN_38, GN_39",
```
Suggest same fix here as well.

pryans replied:
Seems like switching to just numbers would be good here. Simplifies the docs to just a number range.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -108,147 +87,336 @@ function cheat:teleport_to(line)
+}
+
+cheat:createCommand("cheat_tp_hl", "cheat:tp_hl(%line)", cheat.cheat_tp_hl_args,
+"Teleports the player to the given place. Supported places (case insensitive):\n$8GERTRUDE, KUNHUTA, NEUHOF, SAMOPESH",
```
Suggest same fix here as well.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -108,147 +87,336 @@ function cheat:teleport_to(line)
+-- cheat_tp_ml (MILLS)
+-- ============================================================================
+cheat.cheat_tp_ml_args = {
+  place = function(args,name,showHelp) return cheat:argsGetRequired(args, name, showHelp, "Teleport to herbalist") end,
```
copy/paste error, should be to mill.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -108,147 +87,336 @@ function cheat:teleport_to(line)
+}
+
+cheat:createCommand("cheat_tp_ml", "cheat:tp_ml(%line)", cheat.cheat_tp_ml_args,
+"Teleports the player to the given place. Supported places (case insensitive):\n$8BUDIN, KATZEK, KOHELNITZ, LEDETCHKO, RATTAY",
```
Suggest same fix here as well.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -108,147 +87,336 @@ function cheat:teleport_to(line)
+}
+
+cheat:createCommand("cheat_tp_tvn", "cheat:tp_tvn(%line)", cheat.cheat_tp_tvn_args,
+"Teleports the player to the given place. Supported places (case insensitive):\n$8EAST, GLADE, TALMBERG, WAGONERS, WEST, WHEEL",
```
Suggest same fix here as well.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -260,45 +428,48 @@ cheat.cheat_tp_tr_args = {
 cheat:createCommand("cheat_tp_tr", "cheat:tp_tr(%line)", cheat.cheat_tp_tr_args,
-"Teleports the player to the given place. Supported places (case insensitive):\n$8I, II, III, IV, V, VI, VII, VIII, IX, X, XI, XII, XIII, XIV, XV, XVI, XVII, XVIII, XIX, XX,\n$8XXI, XXII, XIII, XXIV, XXV, AI, AII, AII, AIII, AIV, AV",
-"Example", "cheat_tp_tr place:XXV")
+"Teleports the player to the given place. Supported places (case insensitive):\n$8TR_1, TR_2, TR_3, TR_4, TR_5, TR_6, TR_7, TR_8, TR_9, TR_10, TR_11, TR_12, TR_13, TR_14, TR_15,\n$8TR_16, TR_17, TR_18, TR_19, TR_20, TR_21, TR_22, TR_23, TR_24, TR_25, ATR_1, ATR_2, ATR_3, ATR_4, ATR_5",
```
Suggest same fix here as well.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -260,45 +428,48 @@ cheat.cheat_tp_tr_args = {
+  
+  -- All treasure locations. I preferred arabic numerals to roman
+  -- numerals because it is much easier to visit all of them in order.
+  -- (You normally only need to change one digit and re-run the command)
```
I think removing the prefix would make this faster to type. Also why TR_ and ATR_ ?

wilecoyote replied:
Agreed. I will change the description of all commands so that we can shorten them to numbers only.

Since we are in the process of simplifying things it would be nice to have all the teleport commands work like **cheat_teleport_to**. With this command the argument **place:** is not a requirement. However, I don't like the fact that this is the only command which is defined as system command. Since my LUA knowledge is rudimentary, it would be extremely helpful if you could tell me what needs to be changed to make all teleport commands work without this requirement but have them running in the context of cheat and not as system command.

I guess 
```lua
local args = cheat:argsProcess(line, cheat.cheat_tp_bh_args)
```
and
```lua
cheat.cheat_tp_hl_args = {
  place = function(args,name,showHelp) return cheat:argsGetRequired(args, name, showHelp, "Teleport to herbalist") end,
}
```
needs to be changed and also the checking
``` lua
if gender ~= 2 then
    if not nplaceErr then
      if places[cheat:toUpper(nplace)] ~= nil then
        cheat:teleport(places[cheat:toUpper(nplace)])
      else
        cheat:logError("Invalid Place - See list of supported places type: 'cheat_tp_cb ?'")
      end
    end
  else
    cheat:logError("You can't use this command while playing Thereza!")
  end
```
If I would adjust all commands like **cheat_teleport_to** it is clear to me what needs to be done. But as I said before I would prefer to keep the commands in the context of cheat and not turn them into system commands.

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -312,13 +483,49 @@ function cheat:tp_tr(line)
+-- cheat_tp_wc (WOODCUTTER CAMP)
+-- ============================================================================
+cheat.cheat_tp_wc_args = {
+  place = function(args,name,showHelp) return cheat:argsGetRequired(args, name, showHelp, "Teleport to tavern") end,
```
copy/paste error, should be to woodcutter camp

pryans commented on Source/Scripts/cheat_core_player.lua:
```diff
@@ -312,13 +483,49 @@ function cheat:tp_tr(line)
+}
+
+cheat:createCommand("cheat_tp_wc", "cheat:tp_wc(%line)", cheat.cheat_tp_wc_args,
+"Teleports the player to the given place. Supported places (case insensitive):\n$8LEDETCHKO, RATTAY, RATTAY_WOODS, TALMBERG, UZHITZ",
```
Suggest same fix here as well.

wilecoyote committed:
Implemented changes suggested by pryans

wilecoyote committed:
Corrected formatting of cheat_resurrect_npc

wilecoyote committed:
Fine tuned item category ids.
Removed ids which shouldn't be used

wilecoyote committed:
Another item category id which shouldn't be used

wilecoyote committed:
Had to re-add item category ids after testing

wilecoyote committed:
Changed invisible and immortal buff to non-system
Added excludes to cheat_find_buff
Added tp command for easter eggs
Added a lot of tp locations

pryans committed:
Merge branch 'master' into master
